src/main/java/redis/clients/jedis/JedisFactory.java

```diff
@@ -24,7 +24,7 @@
 
   private static final Logger logger = LoggerFactory.getLogger(JedisFactory.class);
 
-  private final AtomicReference<HostAndPort> hostAndPort = new AtomicReference<>();
+  private final AtomicReference<JedisSocketFactory> jedisSocketFactory = new AtomicReference<>();
```

Comment:
It seems we don't have any more constructor where JedisSocketFactory is not being set.
```suggestion
  private final JedisSocketFactory jedisSocketFactory;
```

Reply:
We do 😢 , specifically these three:

```java
/**
 * {@link #setHostAndPort(redis.clients.jedis.HostAndPort) setHostAndPort} must be called later.
 */
protected JedisFactory(final int connectionTimeout, final int soTimeout, final int infiniteSoTimeout,
    final String user, final String password, final int database, final String clientName) {
  this(connectionTimeout, soTimeout, infiniteSoTimeout, user, password, database, clientName, false, null, null, null);
}
```

```java
/**
 * {@link #setHostAndPort(redis.clients.jedis.HostAndPort) setHostAndPort} must be called later.
 */
protected JedisFactory(final int connectionTimeout, final int soTimeout, final int infiniteSoTimeout,
    final String user, final String password, final int database, final String clientName, final boolean ssl,
    final SSLSocketFactory sslSocketFactory, final SSLParameters sslParameters, final HostnameVerifier hostnameVerifier) {
  this(DefaultJedisClientConfig.builder().connectionTimeoutMillis(connectionTimeout)
      .socketTimeoutMillis(soTimeout).blockingSocketTimeoutMillis(infiniteSoTimeout).user(user)
      .password(password).database(database).clientName(clientName)
      .ssl(ssl).sslSocketFactory(sslSocketFactory)
      .sslParameters(sslParameters).hostnameVerifier(hostnameVerifier).build());
}
```

```java
/**
 * {@link #setHostAndPort(redis.clients.jedis.HostAndPort) setHostAndPort} must be called later.
 */
protected JedisFactory(final JedisClientConfig clientConfig) {
  this.clientConfig = clientConfig;
}
```

Reply:
Ahh, I missed the last one.

Reply:
How about a new constructor?

    DefaultJedisSocketFactory(JedisClientConfig config)

Reply:
Got you, so we basically move the NPE to the socket factory instead the jedis factory, I think that makes sense, will update.